src/App.tsx

```diff
@@ -1,24 +1,248 @@
+} from './api/todos';
+import { TodoApp } from './components/TodoApp/TodoApp';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
+import { TempTodo } from './components/TempTodo/TempTodo';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
 
-const USER_ID = 0;
+const USER_ID = 11528;
 
 export const App: React.FC = () => {
+  const [todos, setTodo] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TodosFilter>(TodosFilter.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [request, setRequest] = useState(true);
+  const [loadingId, setLoadingId] = useState<number[]>([]);
+  const [isLoaderActive, setIsLoaderActive] = useState(false);
+  const [todoTitle, setTitle] = useState('');
+
+  const activeTodosCount = todos
+    .filter(({ completed }) => completed === false).length;
+
+  let completedTodosCount = todos
+    .some(({ completed }) => completed === true);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(createdTodo => {
+        setTodo(createdTodo);
+        setRequest(false);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setRequest(false);
+      });
+  }, []);
+
+  const idTimerRef = useRef<number>(0);
+
+  useEffect(() => {
+    if (idTimerRef.current) {
+      window.clearTimeout(idTimerRef.current);
+    }
+
+    idTimerRef.current = window.setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }, [setErrorMessage]);
+
+  const handleAddTodos = (newTodo: Omit<Todo, 'id'>) => {
+    setLoadingId([0]);
+    setIsLoading(true);
+    setRequest(true);
+    addTodos(newTodo)
+      .then((createdTodo) => {
+        setTodo((currentTodo) => [...currentTodo, createdTodo]);
+        setTitle('');
+        setIsLoading(false);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        setTimeout(() => setErrorMessage(''), 3000);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setIsLoading(false);
+        setRequest(false);
+      });
+    const temp: Todo = Object.assign(newTodo, { id: 0 });
+
+    setTempTodo(temp);
+  };
+
+  const handleDelete = (todoId: number) => {
+    setLoadingId([todoId]);
+    setIsLoaderActive(true);
+    deleteTodos(todoId)
+      .then(() => {
+        setTodo(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+        setErrorMessage('');
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete todo');
+      })
+      .finally(() => {
+        setLoadingId([]);
+        setIsLoading(false);
+        setIsLoaderActive(false);
+      });
+  };
+
+  const handleClearCompleted = () => {
+    const deletePromises = todos
+      .filter(({ completed }) => completed)
+      .map(({ id }) => deleteTodos(id));
```

Comment:
Here you can use `reduce` and walk through the array once instead of twice, which will improve performance

Reply:
> Here you can use `reduce` and walk through the array once instead of twice, which will improve performance

Not fixed without any reply